state/modelgeneration.go

```diff
@@ -424,6 +422,52 @@ func (g *Generation) Commit(userName string) (int, error) {
 	return newGenId, nil
 }
 
+// assignedWithAllUnits generates a new value for the branch's
+// AssignedUnits field, to indicate the all units of changed applications
+// are tracking the branch.
+func (g *Generation) assignedWithAllUnits() (map[string][]string, error) {
+	assigned := g.AssignedUnits()
+	for app := range assigned {
+		units, err := appUnitNames(g.st, app)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		assigned[app] = units
+	}
+	return assigned, nil
+}
+
+// commitConfigTxnOps iterates over all the applications with configuration
+// deltas, determines their effective new settings, then gathers the
+// operations representing the changes so that they can all be applied in a
+// single transaction.
+func (g *Generation) commitConfigTxnOps() ([]txn.Op, error) {
+	var ops []txn.Op
+	for appName, delta := range g.Config() {
```

Comment:
Where is the op that inserts the commit into the history and removes the branch?

Reply:
See my comment regarding generation ID above. Once completed it is a model generation and is no longer "in-flight".